### ISSUE TITLE
Why if the input query is same, but the result is total difference when do search at twice?

### ISSUE THREAD

**Issue by dtfinder** [guxd/deep-code-search#21]
Hi, 
I do the search twice at the same input query string, but the result is different. 
My input is the first string from the the train.desc.txt , I think the result may similar as the first index at the train.xxx.txt, but the result is not. even if you search two times, the result it is total different. From my understanding, if model fixed, and input is fixed, the result shall be fixed.

The following is copy of test.

Input Query: determine whether the specified name has been used as a key in this table or any of its parents
How many results? 5
('public int getTransportTypeId ( ) { return Integer . parseInt ( transport_type_id ) ; } \n', 0.23941047)

('public Long ( java . lang . String name , InputStream i ) throws IOException , FormatException { this ( name , new DataInputStream ( i ) . readLong ( ) ) ; } \n', 0.29885766)

('public YearMonth withYear ( int year ) { int [ ] newValues = getValues ( ) ; newValues = getChronology ( ) . year ( ) . set ( this , YEAR , newValues , year ) ; return new YearMonth ( this , newValues ) ; } \n', 0.26326206)

('public static final Token newToken ( int ofKind ) { switch ( ofKind ) { default : return new Token ( ) ; } } \n', 0.24075004)

('public static int osisIdToVerseNum ( String osisID ) { if ( osisID != null ) { String [ ] parts = osisID . split ( "|." ) ; if ( parts . length > 1 ) { String verse = parts [ parts . length - 1 ] ; return Integer . valueOf ( verse ) ; } } return 0 ; } \n', 0.23932476)
Input Query: determine whether the specified name has been used as a key in this table or any of its parents
How many results? 5
('protected boolean hasUncompressedStrip ( ) { return mStripBytes . size ( ) != 0 ; } \n', 0.22734238)

('public static < T > Predicate < T > or ( final Predicate < T > ... that ) { return new OrPredicate < T > ( that ) ; } \n', 0.25138146)

('public Object visit ( ASTPrimaryExpression node , Object data ) { node . childrenAccept ( this , data ) ; return data ; } \n', 0.24480948)

('public static int randColor ( ) { return randColor ( new Random ( ) ) ; } \n', 0.22334205)

('@ Override public void close ( ) throws IOException { super . close ( ) ; __socket . close ( ) ; } \n', 0.2098054)

**Comment by guxd** [comment]
We didn't observe such a phenomenon before.  
Could you double check which part introduces the randomness? Print the intermediate results for the same query and show which part the variation comes from.

**Comment by dtfinder** [comment]
Hi, 

   def search(self,model,query,n_results=10):
        desc=sent2indexes(query, self.vocab_desc)#convert desc sentence into word indices
        desc = np.expand_dims(desc, axis=0)
        desc=gVar(desc)
        desc_repr=model.desc_encoding(desc).data.cpu().numpy()
   

I noticed the desc_repr  value is different from the same input query value. 

Maybe you can check more deep reason for this issue.

**Comment by guxd** [comment]
maybe due to the dropout during prediction.
You can call model.eval() before desc_encoding()

**Comment by guxd** [comment]
I have added model.eval() in the latest revision. Please let me know if the problem remains

**Comment by dtfinder** [comment]
Hi, 
I noticed some error on the torch version code, This time I decided to follow README.md without any code change.
Train:
I used latest version of codesercher.py ,  and set "reload: 0" for configs.py, and  re-train the model with default data.  at this step it seems ok,.

Code Embedding:
I set "relaod:1990" for configs.py

When I run: python codesearcher.py --mode repr_code
I get the following error:
Build Model
Traceback (most recent call last):
  File "codesearcher.py", line 279, in <module>
    codesearcher.load_model(model, conf['reload'])
NameError: name 'codesearcher' is not defined

I change "codesearcher" to "searcher" at line 279, and Code Embedding seems ok.

Search:
I run: python codesercher.py --mode search 
It seems OK, the repeat query with same input is still same.
